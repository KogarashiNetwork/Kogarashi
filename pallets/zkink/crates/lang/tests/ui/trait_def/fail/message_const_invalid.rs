use ink_lang as ink;

#[ink::trait_definition]
pub trait TraitDefinition {
    #[ink(message)]
    const fn message_ref(&self);

    #[ink(message)]
    const fn message_mut(&mut self);
}

fn main() {}
