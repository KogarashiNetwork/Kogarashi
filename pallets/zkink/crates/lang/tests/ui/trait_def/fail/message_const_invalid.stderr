error: const ink! trait methods are not supported
 --> tests/ui/trait_def/fail/message_const_invalid.rs:6:5
  |
6 |     const fn message_ref(&self);
  |     ^^^^^
